echopype/convert/set_groups_ek60.py

```diff
@@ -310,6 +310,16 @@ def set_platform(self) -> xr.Dataset:
         ds = xr.merge([ds, ds_plat], combine_attrs="override")
         ds = ds.assign_attrs(platform_dict)
 
+        # If `.IDX` file exists and `.IDX` data is parsed
+        if (
+            (self.parser_obj.idx_file != "")
+            and self.parser_obj.idx["vessel_distance"]
+            and self.parser_obj.idx["latitude_idx"]
+            and self.parser_obj.idx["longitude_idx"]
+            and self.parser_obj.idx["timestamp"]
+        ):
```

Comment:
I think this is fine, but I am curious about if the later checks (those involving `self.parser_obj.idx`) are included to catch cases where `self.parser_obj.idx_file != ""` is satisfied but `self.parser_obj.idx` does not contain any data?

Reply:
Yup, that's what it's meant to catch. In the case where the parser reads nothing from the IDX file